authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,754 @@
+Where `<y>` and `<z>` are the numbers corresponding to the version of {product-title} you are installing.
+
+. Create the `install-config.yaml` file:
++
+[source,terminal]
+----
+$ ./openshift-install create install-config
+----
+
+. Configure the cluster to install with the CCO in manual mode:
++
+[source,terminal]
+----
+$ echo "credentialsMode: Manual" >> install-config.yaml
+----
+
+. Create install manifests:
++
+[source,terminal]
+----
+$ ./openshift-install create manifests
+----
+
+[id="sts-mode-installing-preflight-using"]
+==== Using the sts-preflight tool
+
+.Prerequisites
+* Download the link:https://github.com/sjenning/sts-preflight[`sts-preflight` tool].
+
+.Procedure
+
+. Using an AWS account with administrator-level permissions, run the following command to:
++
+--
+** Generate public and private `ServiceAccount` object signing keys.
+** Create an S3 bucket.
+** Upload the OIDC config into the S3 bucket.
+** Set up an IAM Identity Provider that trusts the S3 bucket.
+** Create an IAM role with administrator rights to use for installation.
+** Create IAM roles for each AWS `CredentialsRequest` CR.
++
+[source,terminal]
+----
+$ ./sts-preflight create --infra-name <aws-infra-name> --region <aws-region> --credentials-requests-to-roles <path-to-credreqs.yaml>
+----
++
+Where `<aws-infra-name>` is a prefix the `sts-preflight` tool uses when naming the objects that are created in AWS.
+--
++
+The files that are required by the installer are added to the `_output` directory.
+
+. The `sts-preflight` tool does not put conditions that allow only specific `ServiceAccount` objects to assume a role on the IAM roles that it creates. To restrict which identities can assume which role, you must manually modify the trust relationship of each role to add a corresponding condition.
++
+[NOTE]
+====
+Role names that exceed 64 characters might be truncated.
+====
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-image-registry-installer-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-image-registry:registry",
+      "system:serviceaccount:openshift-image-registry:cluster-image-registry-operator"
+    ]
+  }
+}
+----
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-ingress-operator-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-ingress-operator:ingress-operator"
+    ]
+  }
+}
+----
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-cluster-csi-drivers-ebs-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
```

Comment:
the same as above: `"s3.<aws-region>.amazonaws.com/<aws-infra-name>-installer:sub"`